src/agent/coverage/src/binary.rs

```diff
@@ -124,14 +124,16 @@ pub fn find_coverage_sites(
             continue;
         }
 
-        if let Some(location) = symcache.lookup(function.offset.0).next() {
-            if let Some(file) = location.file() {
-                let path = file.full_path();
-
-                if allowlist.source_files.is_allowed(path) {
-                    let blocks =
-                        block::sweep_region(module, &debuginfo, function.offset, function.size)?;
-                    offsets.extend(blocks.iter().map(|b| b.offset));
+        let blocks = block::sweep_region(module, &debuginfo, function.offset, function.size)?;
+
+        for block in &blocks {
+            if let Some(location) = symcache.lookup(block.offset.0).next() {
+                if let Some(file) = location.file() {
+                    let path = file.full_path();
+
+                    if allowlist.source_files.is_allowed(&path) {
```

Comment:
Add a comment, and apply the function allowlist here, too.

This may result in reporting artifacts around inlined calls (the `call` lines may appear unreachable). However, we don't have easy access to the full inline stack, so this is best for now (to avoid the worse artifact of allowlists "not working").